Fb globus 30093

cnathe opened the pull request.

labkey-susanh committed:
Spec 30093: Globus file transfer - add new module properties

labkey-susanh committed:
Spec 30093: add new module properties for client id and secret; add new link in file transfer web part; integrate with oauth2 library to get; add auth action and stub of transfer action

labkey-susanh committed:
Spec 30093: use web part properties and add a web part factory to allow multiple transfer web parts on the page

labkey-susanh committed:
Spec 30093: add UI for file transfer preparation; get data stored in session; add stub of transfer action and flesh out prepare action

labkey-susanh committed:
add jars.txt for new dependency on google oauth client

labkey-susanh committed:
Issue 27659: Update jars.txt for Gradle build; add gitignore directive for dependencies.txt file

labkey-susanh committed:
Spec 30093: Wire up the transfer request calls. Updates to UI for file transfer preparation page.  Query Globus API for endpoint metadata to display.

labkey-susanh committed:
remove obsolete .iml file

labkey-susanh committed:
Spec 30093: move configurations from module properties to admin console settings groups

labkey-susanh committed:
Spec 30093: remove use of WebDav; move source endpoint configuration from Module property to admin settings;

labkey-susanh committed:
Spec 30093: make local file directory relative

labkey-susanh committed:
Spec 30093: make admin setting for root directory associated with an endpoint; validate the web part relative directory when displaying the web part

labkey-susanh committed:
Spec 30093: remove TODOs; add CSRF where applicable

labkey-susanh committed:
Spec 30093: get the file transfer provider from the context (web part properties)

labkey-susanh committed:
Issue 30452: don't assume an integer key for the selection list

labkey-susanh committed:
Spec 30093: Update QuerySchema so we can pull in the appropriate webPart properties for customizing grids
Issue 30452: check for both Integer and auto-increment integer key field type

labkey-susanh committed:
Issue 30455: update wording

labkey-susanh committed:
Issue 30453: update Transfer link to user the root container instead of the current container in the return URL

labkey-susanh committed:
Issue 30466: display transfer label in messaging

cnathe commented:
Creating pull request for code review.

cnathe commented on src/org/labkey/filetransfer/FileTransferController.java:
```diff
@@ -53,63 +77,71 @@ public FileTransferController()
     }
 
+    @CSRF
     @RequiresPermission(AdminOperationsPermission.class)
```
It looks like the success URL returns the user to the Admin Console. Is this now an admin console only action (i.e. root container only)? If so, should probably add the @AdminConsoleAction annotation, and think about if this set of properties should be read-only, hidden, or update-able for Application Admins.

labkey-susanh replied:
Good call.  Can you remind me how I specify what Application Admins can do?

cnathe replied:
See PermissionCheckableAction.java around line 163.

Using @AdminConsoleAction(AdminOperationsPermission.class) = container must be root, user must have AdminReadPermission to GET, user must have AdminOperationsPermission to POST.

Using just @AdminConsoleAction = container must be root, user must have AdminReadPermission to GET, user must have AdminPermission to POST.

And if you want to restrict the GET as well as the POST for an admin console action, use @AdminConsoleAction @RequiresPermission(AdminOperationsPermission.class).

cnathe commented on src/org/labkey/filetransfer/FileTransferController.java:
```diff
@@ -119,10 +151,362 @@ public URLHelper getSuccessURL(FileTransferConfigForm form)
+    @RequiresPermission(ReadPermission.class)
+    public class TokensAction extends RedirectAction<AuthForm>
+    {
+        Boolean authorized = true;
```
The default for this param is TRUE? I would have expected it to be NULL or FALSE, but I could be wrong.

labkey-susanh replied:
Yeah, should be false.  Changed.

cnathe commented on src/org/labkey/filetransfer/FileTransferController.java:
```diff
@@ -119,10 +151,362 @@ public URLHelper getSuccessURL(FileTransferConfigForm form)
+
+                // TODO check if this actually retrieves from the database or if there's more going on
+//                StoredCredential credential = store.get(null);
+//                if (credential.getAccessToken() == null || credential.getExpirationTimeMilliseconds() == null || credential.getExpirationTimeMilliseconds() <= 0)
```
Should this commented out code be removed, or is it something to be handled in a future sprint?

labkey-susanh replied:
Has been removed.

cnathe commented on src/org/labkey/filetransfer/FileTransferManager.java:
```diff
@@ -54,39 +78,105 @@ public static FileTransferManager get()
+                {
+                    selectionIds.add(Integer.parseInt(val));
+                }
+                filter = new SimpleFilter(new SimpleFilter.InClause(FieldKey.fromParts(listDef.getKeyName()), selectionIds));
```
Not needed for this story, but I wonder if this "return me my selected keys of type X" helper would be useful in other places.

labkey-susanh replied:
Yes, I would think so.  I certainly looked for that when doing this wrong a couple of ways.  I'll check if that's easy to do somewhere and otherwise leave a TODO.

cnathe commented on src/org/labkey/filetransfer/FileTransferModule.java:
```diff
@@ -66,31 +64,23 @@ public boolean hasScripts()
 
+    public static void registerAdminConsoleLinks()
+    {
+        AdminConsole.addLink(AdminConsole.SettingsLinkType.Configuration, "File Transfer", FileTransferController.getComplianceSettingsURL());
```
I believe you can pass another param to the end of the addLink() call to set the permission level needed to see this link. I believe this action is currently set to only be operations only (i.e. site-admin). In that case, does the page currently show an error for the ApplicationAdmin?

cnathe commented on src/org/labkey/filetransfer/model/TransferEndpoint.java:
```diff
@@ -0,0 +1,66 @@
+        this.displayName = displayName;
+    }
+
+    public void setDisplay_name(String displayName) { setDisplayName(displayName); }
```
Is this just an artifact or is this actually used by some part of the process? Just curious.

labkey-susanh replied:
The query to the Globus API returns values with underscores between words (Python-style), do this setter is used when deserializing from that call.  We display the endpoint name on the fileTransfer page (if we have it).

cnathe commented on src/org/labkey/filetransfer/view/fileTransferConfig.jsp:
```diff
@@ -16,31 +16,26 @@
     @Override
     public void addClientDependencies(ClientDependencies dependencies)
     {
-        dependencies.add("Ext4");
```
This file / form still uses Ext4 right? If so, why was the dependency removed here?

labkey-susanh replied:
Good question.  I'm not sure why that happened, but it's been added back in now.

cnathe commented on src/org/labkey/filetransfer/view/fileTransferConfig.jsp:
```diff
@@ -52,115 +47,149 @@
+            width: 535,
+            padding: '10px 0 0 25px',
             hidden: false,
             disabled: false,
```
I don't have this branch pulled down locally to look at the UI, but isn't "false" the default for the hidden and disabled properties of the Ext4 form elements? If so, can probably remove from all of the items below. I don't see anywhere that updates the hidden or disabled state of an item after it is initialized. Probably an artifact of the previous set of form items.

cnathe commented on src/org/labkey/filetransfer/view/fileTransferWebPartConfig.jsp:
```diff
@@ -0,0 +1,238 @@
+    @Override
+    public void addClientDependencies(ClientDependencies dependencies)
+    {
+        dependencies.add("sqv");
```
This should also explicitly declare its dependency on Ext4.

labkey-susanh replied:
Nick indicated when randomly looking over my shoulder that it wasn't necessary since sqv pulls in Ext4, but it is better to declare the dependency explicitly since we use Ext even without sqv.

cnathe commented on src/org/labkey/filetransfer/view/fileTransferWebPartConfig.jsp:
```diff
@@ -0,0 +1,238 @@
+            width: 510,
+            padding: '10px 0 25px 0',
+            hidden: false,
+            disabled: false,
```
Same comment as above regarding the hidden and disabled props.

labkey-susanh replied:
Done

labkey-susanh committed:
Issue 30468: catch unknown host excetpoin and log an error if the HTTP GET request fails
Spec 30093: update permissions for Admin Console page so Operations permission is required, but Application Admin can see the values; add explicit dependency on Ext4 for JSPs

labkey-susanh committed:
Issue 30468: catch exception when client secret is incorrect and display better error message.

labkey-susanh committed:
Issue 30471: update error message if unable to retrieve submission id when source endpoint is misconfigured.

labkey-susanh committed:
Spec 30093: display default values for service URLs

labkey-susanh committed:
Spec 30093: update permissions and container used for credentials

labkey-susanh committed:
Spec 30093: update return URL to not use the root container

labkey-danield committed:
#30093 Globus File Transfer Test Automation. Added a Admin page for file transfer, replaced the customize page for webpart. Updated sample data to be more inline with ITN files.

labkey-danield committed:
#30093 Globus File Transfer Test Automation update based on test code review feedback

labkey-susanh committed:
remove TokensAction from read permissions assertion

labkey-susanh committed:
Merge remote-tracking branch 'origin/fb_globus_30093' into fb_globus_30093
